src/read/read_ref.rs

```diff
@@ -39,6 +39,22 @@ where
         }
     }
 
+    /// Iterate over all keys in the map.
+    ///
+    /// Be careful with this function! While the iteration is ongoing, any writer that tries to
+    /// refresh will block waiting on this reader to finish.
+    pub fn keys(&self) -> impl Iterator<Item = &K> {
+        self.iter().map(|(k, _)| k)
+    }
+
+    /// Iterate over all values in the map.
+    ///
+    /// Be careful with this function! While the iteration is ongoing, any writer that tries to
+    /// refresh will block waiting on this reader to finish.
+    pub fn values(&self) -> impl Iterator<Item = &V> {
```

Comment:
One more question - should `values()` return an iterator over `&V` or over `Values<V, ...>`. That is, should be a flat iterator over each value or should it be an iterator over each value set? If this were `HashMap<K, Vec<V>>` then I'd expect over each value set but since you don't get the choice of not having the value collection I didn't know what the "expected" value type is.

Reply:
I think it should be an iterator over the `Values<V>`. The user can always choose to flatten with `Iterator::flatten()`.